twpca/model.py

```diff
@@ -178,25 +180,27 @@ def __init__(self, data, n_components,
             # trial i, time j, factor k, neuron n
             self._pred = tf.einsum('ijk,nk->ijn', self._warped_time_factors, self._params['neuron'])
 
-        # objective function (note that nan values are zeroed out by self._mask)
-        self._regularization = tf.reduce_sum([self._regularizers[k](self._params[k]) for k in self._params.keys()])
-        self._recon_cost = tf.reduce_sum(self._mask * (self._pred - self._data)**2) / self.num_datapoints
-        self._objective = self._recon_cost + self._regularization
+        # objective function (note that nan values are zeroed out by _mask)
+        regularization = tf.reduce_sum([regularizers[k](self._params[k])
+                                        for k in self._params.keys()])
+        recon_cost = tf.reduce_sum(_mask * (self._pred - self._data)**2) / num_datapoints
+        self._objective = recon_cost + regularization
```

Comment:
I'm not sure if you guys use `recon_cost` and `regularization`, but if you do, we can keep them as class attributes ...